### ISSUE TITLE
PolyComm is too generic

### ISSUE THREAD

**Issue by mimoo** [o1-labs/proof-systems#409]
The way PolyComm is used makes it really hard to parse the code.

```rust
pub struct PolyComm<C>
{
    pub unshifted: Vec<C>,
    pub shifted: Option<C>,
}
```

* it's used to represent a non-hidden commitment part, with `C` a curve point
* it's used to represent the blinding of a commitment, with `C` a field element
* it's used with chunked polynomials, with `C` a `Vec<C>`

Ideally, there should be a struct that represents more clearly a blinded commitment from a non-blinded commitment.

Targeting chunked/split polynomials is harder as we don't have good abstractions for that.

**Comment by mimoo** [comment]
(Note that it would be annoying to change this due to the ocaml bindings)

**Comment by xiangxiecrypto** [comment]
Could take this, but i think should take some time to be familiar with the codes.

**Comment by mimoo** [comment]
I think creating a `BlindedPolyComm` struct would be a good idea. If you can find in the code where this is used, you can check the `mask()` function in commitment.rs to see how it creates the blinding part (the second part of the tuple)

**Comment by xiangxiecrypto** [comment]
> 

What `unshifted` and `shifted` mean in `PolyComm`?

Basically, we could let `PolyComm` denote  the non-blinding one, contains curve points ` <a, G>`.
Let `BlindedPolyComm` denote the blinding one, contains `(<a, G>+ wH, w)`, which is a curve point and a field point.

Does it make sense?

**Comment by mimoo** [comment]
If I remember correctly, shifted is used when you want to bound the degree of a polynomial. 

So imagine that you have a polynomial like `[c1, c2, c3, c4]` (so f = c1 + c2 x + c3 x^3 + c4 x^4).
Let's imagine that you can only commit to polynomials of size 2 (degree 1), that means you would commit `C1, r1 = commit([c1, c2])` and `C2, r2 = commit([c3, c4])` separately (where C_i is a commitment, and r_i is the blinding factor that needs to remain secret, see [mina book](https://o1-labs.github.io/mina-book/crypto/plonk/commitments.html)).
You'd get a `PolyComm` where `unshifted` is a vector containing `[C1, C2]` (and `[r1, r2]` would be contained in a `PolyComm<F>`).

If you wanted to make sure that the degree is actually 2, not 3 (so c4 has to be 0) then you would shift your commitment to `[c3, c4]` as a commitment to `[0, c3]`. So your PolyComm would have a `unshifted = C1` and `shifted = commit([0, c3])`. I wrote more about that trick here: https://o1-labs.github.io/mina-book/crypto/plonk/inner_product_api.html#enforcing-an-upperbound-on-the-polynomial-degree

**Comment by xiangxiecrypto** [comment]
Just reading the mina book here: https://o1-labs.github.io/mina-book/specs/polynomial_commitment.html.
I think you considered this issue before, but it looks like the code does not follow this specs? :)

**Comment by mimoo** [comment]
yeah I would disregard this spec as it isn't really complete

**Comment by xiangxiecrypto** [comment]
> BTW, disclaimer that this would be a change that would have some annoying repercussion on Mina and so probably wouldn't be fit for the near term. Still, it'd be good to use this issue as brainstorming to see if changing this structure is worth it at some point. (I believe it will be.)

So, I think we could leave it for the future work, and handling #432 first maybe a good choice? Since chunked polynomials also involve in using `PolyComm`.

**Comment by mimoo** [comment]
Yeah I think it'd be a better start. Especially as some of #432 can be created in isolation in o1-utils even before we start using it in kimchi :o
